src/test/java/org/jabref/model/entry/identifier/DOITest.java

```diff
@@ -16,22 +16,35 @@ public void acceptPlainDoi() {
         assertEquals("10.1002/(SICI)1522-2594(199911)42:5<952::AID-MRM16>3.0.CO;2-S",
                 new DOI("10.1002/(SICI)1522-2594(199911)42:5<952::AID-MRM16>3.0.CO;2-S").getDOI());
         assertEquals("10.1126/sciadv.1500214", new DOI("10.1126/sciadv.1500214").getDOI());
+        // Short DOI
+        assertEquals("10/gf4gqc", new DOI("10/gf4gqc").getDOI());
+    }
+
+    @Test
+    public void rejectInvalidShortDoi() {
+        assertThrows(IllegalArgumentException.class, () -> new DOI("20/abcd"));
     }
 
     @Test
     public void ignoreLeadingAndTrailingWhitespaces() {
         assertEquals("10.1006/jmbi.1998.2354", new DOI("  10.1006/jmbi.1998.2354 ").getDOI());
+        // Short DOI
+        assertEquals("10/gf4gqc", new DOI("   10/gf4gqc ").getDOI());
     }
 
     @Test
     public void rejectEmbeddedDoi() {
         assertThrows(IllegalArgumentException.class, () -> new DOI("other stuff 10.1006/jmbi.1998.2354 end"));
+        // Short DOI
```

Comment:
Please create a new test for the short doi (each test should only contain one assertion - or at least only assert that one specific feature works as intended).

Reply:
Good call. Extracted all tests related to short DOIs to another tests. I also fixed not handling upper case letters in short DOIs.